realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -59,6 +59,9 @@
  */
 public final class RealmResults<E extends RealmObject> extends AbstractList<E> {
 
+    // JNI layer will return -1 if core returns realm::not_found
+    private static final long NOT_FOUND = -1;
```

Comment:
This constant is already in TableOrView where it is called NO_MATCH. `NOT_FOUND` is probably a better name though.